internal/integration/api/common.go

```diff
@@ -168,6 +169,58 @@ func (suite *CommonSuite) TestDNSResolver() {
 	suite.Require().Contains(stderr, "'index.html' saved")
 }
 
+// TestDNSResolverLongNames verifies that external DNS resolving of the long url names works from a pod.
+func (suite *CommonSuite) TestDNSResolverLongNames() {
+	if suite.Cluster != nil {
+		// cluster should be healthy for kube-dns resolving to work
+		suite.AssertClusterHealthy(suite.ctx)
+	}
+
+	const (
+		namespace = "default"
+		pod       = "dns-test"
+	)
+
+	_, err := suite.Clientset.CoreV1().Pods(namespace).Create(suite.ctx, &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pod,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  pod,
+					Image: "alpine",
+					Command: []string{
+						"tail",
+						"-f",
+						"/dev/null",
+					},
+				},
+			},
+			TerminationGracePeriodSeconds: pointer.To[int64](0),
+		},
+	}, metav1.CreateOptions{})
+
+	suite.Require().NoError(err)
+
+	defer func(req *require.Assertions) {
+		err = suite.Clientset.CoreV1().Pods(namespace).Delete(suite.ctx, pod, metav1.DeleteOptions{})
+		req.NoError(err)
+	}(suite.Require())
+
+	// wait for the pod to be ready
+	suite.Require().NoError(suite.WaitForPodToBeRunning(suite.ctx, time.Minute, namespace, pod))
+
+	// Twitch subdomain for video streaming
+	stdout, stderr, err := suite.ExecuteCommandInPod(suite.ctx, namespace, pod, "nslookup video-edge-3e7abd.pdx01.abs.hls.ttvnw.net")
```

Comment:
can we merge into the previous test? so we don't schedule two pods?

Reply:
Done!